src/main/java/sirius/db/mixing/properties/SQLEntityRefProperty.java

```diff
@@ -81,8 +81,10 @@ protected SQLEntityRefProperty(EntityDescriptor descriptor, AccessPath accessPat
     public void contributeToTable(Table table) {
         table.getColumns().add(new TableColumn(this, Types.BIGINT));
 
-        if (SQLEntity.class.isAssignableFrom(getReferencedType()) && Strings.areEqual(getDescriptor().getRealm(),
-                                                                                      getReferencedDescriptor().getRealm())) {
+        boolean createFkConstraint = SQLEntity.class.isAssignableFrom(getReferencedType()) && Strings.areEqual(
+                getDescriptor().getRealm(),
+                getReferencedDescriptor().getRealm()) && !entityRef.isWeak();
+        if (createFkConstraint) {
```

Comment:
i think maybe instead of a variable , a private method shouldCreateFkConstraint() could look a little more 'clean'